Cors Middleware

albanv opened the pull request.

albanv committed:
feat(cors): add cors middleware

p-j commented on packages/eapi-middleware-cors/src/withCors.ts:
```diff
@@ -1 +1,65 @@
+  origin: string
+}
+
+/**
+ * @param options
+ * @param options.response response on which apply cors headers
+ * @param options.origin origin to set for access-control-allow-origin
+ * @param options.isOriginAllowed an optional function to validate the origin of the request
+ * @returns a Response with CORS headers
+ */
```
This lacks some of the options

p-j commented on packages/eapi-middleware-cors/src/withCors.ts:
```diff
@@ -1 +1,65 @@
+  return function _withCors(requestHandler) {
+    return async function corsHandler({ event, request, params }) {
+      const response = await requestHandler({ event, request, params })
+      const origin = request.headers.get('Origin') || '*'
```
if the `Origin` header is missing, shouldn't we return the response untouched?
There's no point adding CORS headers if there is no Origin, no?

p-j reviewed (approved):
The documentation needs update & the README is missing.
There is the one point about defaulting to '*' versus not adding any header that I'm unsure of.

p-j committed:
chore(types): use the latest types

p-j committed:
feat(cors): handle more CORS headers

- add Access-Control-Allow-Credentials
- add Access-Control-Expose-Headers
- add sensible defaults
- change the behavior when the Origin header is not present

p-j committed:
doc(cors): add README

p-j committed:
chore(pkg): add a lint script
